### ISSUE TITLE
performance gap of reproductive result in GPQA

### ISSUE THREAD

**Issue by Ericmututu** [GAIR-NLP/LIMO#5]
Hi, thanks for sharing such great work!

I attempted to reproduce the results reported in the paper using the official experimental settings. Below are my results (train once, pass@1):

Step | AIME24 | MATH500 | GPQA |
100 | 0.5000 | 0.9280 | 0.5000 |
200 | 0.6000 | 0.9320 | 0.5909 |
300 | 0.5333 | 0.9380 | 0.5758 |
382 | 0.4667 | 0.9380 | 0.5859 |

The results for AIME24 and MATH500 seem reasonable, however, the result for GPQA significantly deviates from the reported results (0.667) in the paper. I am wondering if I might have missed some key techniques, or if it would be possible to share more details about training/evaluation?

BTW, "Beginning with an initial pool of tens of millions of problems" in Page6 (2-th row), I am very curious about the sources of these data (beyond the known NuminaMath-CoT, AIME, MATH).

Thanks in advance!

**Comment by lmc8133** [comment]
How many GPUs do you use?

**Comment by Ericmututu** [comment]
> How many GPUs do you use?

64 x A100

**Comment by lmc8133** [comment]
> > How many GPUs do you use?
> 
> 64 x A100

So do you follow the author's config file? If so, the total steps should be:
817*15(epochs) / (64(GPUs)*1(gradient_accumulation_steps)) = 191.
I'm a little confused

**Comment by Ericmututu** [comment]
> > > How many GPUs do you use?
> > 
> > 
> > 64 x A100
> 
> So do you follow the author's config file? If so, the total steps should be: 817*15(epochs) / (64(GPUs)*1(gradient_accumulation_steps)) = 191. I'm a little confused

tp=2, i.e. 817*15/(64/2)=382

**Comment by Ericmututu** [comment]
> How do you reproduce this result? I tried to eval their model directly on aime24, but just got 33.3 accuracy (greedy generation). It's much lower on my own training model. Could you please provide the evaluation detail? Thanks!

1. I use the same hyperparameters (e.g., lr=5e-6, epoch=15, seqlen=32768) as the paper, but megatron framework used in my experiment.
2. I evaluate my trained model in two ways (based on vllm0.6.6): 
(i) {"max_tokens":32768,"temperature":0.7,"top_p":0.95} -> min_AIME24=0.47 at 382 step (as the top question shown);
(ii) {"max_tokens":32768,"temperature":0} -> min_AIME24=0.36 at 100 step.

**Comment by radna0** [comment]
> > How do you reproduce this result? I tried to eval their model directly on aime24, but just got 33.3 accuracy (greedy generation). It's much lower on my own training model. Could you please provide the evaluation detail? Thanks!
> 
> 1. I use the same hyperparameters (e.g., lr=5e-6, epoch=15, seqlen=32768) as the paper, but megatron framework used in my experiment.
> 2. I evaluate my trained model in two ways (based on vllm0.6.6):
>    (i) {"max_tokens":32768,"temperature":0.7,"top_p":0.95} -> min_AIME24=0.47 at 382 step (as the top question shown);
>    (ii) {"max_tokens":32768,"temperature":0} -> min_AIME24=0.36 at 100 step.

What model was this trained on?

**Comment by Ericmututu** [comment]
> > > How do you reproduce this result? I tried to eval their model directly on aime24, but just got 33.3 accuracy (greedy generation). It's much lower on my own training model. Could you please provide the evaluation detail? Thanks!
> > 
> > 
> > 
> > 1. I use the same hyperparameters (e.g., lr=5e-6, epoch=15, seqlen=32768) as the paper, but megatron framework used in my experiment.
> > 2. I evaluate my trained model in two ways (based on vllm0.6.6):
> >    (i) {"max_tokens":32768,"temperature":0.7,"top_p":0.95} -> min_AIME24=0.47 at 382 step (as the top question shown);
> >    (ii) {"max_tokens":32768,"temperature":0} -> min_AIME24=0.36 at 100 step.
> 
> What model was this trained on? And what was the initial AIME24 score?

Qwen2.5-32B-Instruct, and the initial AIME24 score of Qwen2.5-32B-Instruct is 0.2 in my experiment.

**Comment by radna0** [comment]
So we can do conclude LIMO does help with mathematical scores, many others like OpenR1 has seen the same thing. One interesting thing I want to know is how much it affected other evals for getting better math scores? Did you conduct any other experiments?

Also Does it make sense training more than 382 steps? Or for when temperature was 0, keep training up until 382 steps? Any reason why you didn't do this?

**Comment by Ericmututu** [comment]
> So we can do conclude LIMO does help with mathematical scores, many others like OpenR1 has seen the same thing. One interesting thing I want to know is how much it affected other evals for getting better math scores? Did you conduct any other experiments?
> 
> Also Does it make sense training more than 382 steps? Or for when temperature was 0, keep training up until 382 steps? Any reason why you didn't do this?
> 
> Extra: Have you tested on AIME 2025 Part I?

No more experiments and evaluations have been done yet, and 382 is simply a reproducible choice without much reason ~
:)

**Comment by radna0** [comment]
How long did it take for 384 steps on 64 x A100?

**Comment by Ericmututu** [comment]
> How long did it take for 384 steps on 64 x A100?

~ 5 hours

**Comment by BLeaves** [comment]
Thank you for your interest in our project and for raising these points!

I noticed that you're evaluating models at specific steps (100, 200, 300, and 382). In our case, we conducted a more comprehensive evaluation by **testing all 15 models from each epoch and selected the best performing one** as LIMO (For our published results, this was epoch 9, though this might vary due to random factors).

Also, we used 32 GPUs while your implementation uses 64 GPUs. This difference leads to different batch sizes, which could account for some minor discrepancies in the results.
